File: java/src/main/java/mx/iteso/MyUsers.java
package mx.iteso;

import mx.iteso.privileges.Approve;
import mx.iteso.privileges.Comment;
import mx.iteso.privileges.Delete;
import mx.iteso.privileges.Publish;
import mx.iteso.users.EndUser;

/**
 * Created by Maggie Jauregui on 10/25/19.
 */
public final class MyUsers {

    /**
     * Class constructor.
     */
    private MyUsers() {

    }

    /**
     * Main method to run.
     * @param args input args.
     */
    public static void main(final String[] args) {
        String adminPermissions;

        //Admin user
        EndUser admin = new EndUser("Administrator");

        //Creating permissions for Admin
        Comment adminComment = new Comment(admin);
        Approve adminAprrove = new Approve(admin);
        Delete adminDelete = new Delete(admin);
        Publish adminPublish = new Publish(admin);

        //Giving permissions to Admin
        admin.setPublish(adminPublish.publish());
        admin.setComment(adminComment.comment());
        admin.setDelete(adminDelete.delete());
        admin.setApprove(adminAprrove.approve());

        System.out.println(admin.getName() + " " + admin.getApprove());
        System.out.println(admin.getName() + " " + admin.getDelete());
        System.out.println(admin.getName() + " " + admin.getComment());
        System.out.println(admin.getName() + " " + admin.getPublish());



        //Publisher user
        EndUser publisher = new EndUser("Publisher");

        //Creating permissions for Publisher
        Approve publisherAprrove = new Approve(publisher);
        Publish publisherPublish = new Publish(publisher);

        //Giving permissions to Publisher
        publisher.setPublish(publisherPublish.publish());
        publisher.setApprove(publisherAprrove.approve());

        System.out.println(publisher.getName() + " " + publisher.getApprove());
        System.out.println(publisher.getName() + " " + publisher.getDelete());
        System.out.println(publisher.getName() + " " + publisher.getPublish());
        System.out.println(publisher.getName() + " " + publisher.getComment());



        //Editor user
        EndUser editor = new EndUser("Editor");

        //Creating permissions for Editor
        Comment editorComment = new Comment(editor);
        Publish editorPublish = new Publish(editor);

        //Giving permissions to Editor
        editor.setPublish(editorPublish.publish());
        editor.setComment(editorComment.comment());

        System.out.println(editor.getName() + " " + editor.getApprove());
        System.out.println(editor.getName() + " " + editor.getDelete());
        System.out.println(editor.getName() + " " + editor.getComment());
        System.out.println(editor.getName() + " " + editor.getPublish());



    }


}

File: python/MyUsers.py
from approve import Approve
from delete import Delete
from comment import Comment
from publish import Publish
from EndUser import EndUser

def __init__(self):
    pass

def main():
    admin = EndUser("Admin")
    publisher = EndUser("Publisher")
    editor = EndUser("Editor")

    adminComment = Comment(admin)
    adminApprove = Approve(admin)
    adminPublish = Publish(admin)
    adminDelete = Delete(admin)

    admin.setPublish(adminPublish.publish())
    admin.setComment(adminComment.comment())
    admin.setDelete(adminDelete.delete())
    admin.setApprove(adminApprove.approve())

    print(admin.getName() + ' ' + admin.getDelete())
    print(admin.getName() + ' ' + admin.getApprove())
    print(admin.getName() + ' ' + admin.getPublish())
    print(admin.getName() + ' ' + admin.getComment())

    publisherApprove = Approve(publisher)
    publisherPublish = Publish(publisher)

    publisher.setPublish(publisherPublish.publish())
    publisher.setApprove(publisherApprove.approve())

    print(publisher.getName() + ' ' + publisher.getDelete())
    print(publisher.getName() + ' ' + publisher.getApprove())
    print(publisher.getName() + ' ' + publisher.getPublish())
    print(publisher.getName() + ' ' + publisher.getComment())

    editorComment = Comment(editor)
    editorPublish = Publish(editor)

    editor.setPublish(editorPublish.publish())
    editor.setComment(editorComment.comment())

    print(editor.getName() + ' ' + editor.getDelete())
    print(editor.getName() + ' ' + editor.getApprove())
    print(editor.getName() + ' ' + editor.getPublish())
    print(editor.getName() + ' ' + editor.getComment())

if __name__ == '__main__':
    main()
